util/src/main/java/org/datacommons/util/StatChecker.java

```diff
@@ -192,25 +194,80 @@ private boolean isSvObWithNumberValue(McfGraph.PropertyValues node) {
   private boolean shouldExtractSeriesInfo(McfGraph.PropertyValues node) {
     String placeDcid = McfUtil.getPropVal(node, Vocabulary.OBSERVATION_ABOUT);
     if (placeDcid.isEmpty()) return false;
+
+    // if shouldGenerateSamplePlaces is true, and the bucket that the placekey
+    // belongs has empty space, the sample place will be added to the
+    // samplePlaces map
+    attemptAddingPlaceToSamplePlaces(placeDcid);
```

Comment:
I am on the fence about this, but preferred to break it up because `shouldExtractSeriesInfo` implies read-only, but `attemptAddingPlaceToSamplePlaces` is potentially write  (modifies samplePlaces field of the class). Maybe we should move `attempt...` out of `should...` to avoid having a state-modifying behavior inside a seemingly read-only function?

Reply:
The long name of the function doing some pretty specific logic called from one place from a not-so-long function is a "smell test" that the function should be inlined.

For e.g., the long block of comment at the callsite describing the logic should instead just be inlined code.